app/controllers/DetailsUpdatedController.scala

```diff
@@ -38,9 +38,8 @@ class DetailsUpdatedController @Inject() (
     with I18nSupport
     with Logging {
 
-  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData) {
+  def onPageLoad(fiName: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
```

Comment:
Not entirely convinced of using `fiName` in the URL. This could be better approach or even better alternative available. Let me think a bit.